### DISCUSSION TITLE
Establishing sponsorship for Leaflet 💰

### DISCUSSION THREAD

**Discussion by mourner** [Leaflet/Leaflet#8215] [Announcements]
For the past 11 years of the project's history, it never accepted any direct sponsorships or funding. It was always supported indirectly through companies allowing their employees to contribute to Leaflet at work. I myself never got my hands on establishing proper direct channels for sponsorship and funding — quoting from discussions in #7704: 

> I guess I never got around to actually doing that because I'm quite afraid of taking on the associated responsibility — of managing funds, tracking other contributor's work to distribute them fairly, etc. This seems like a very stressful thing, and I already have so much stress and commitments in my life (full-time work, family, two kids, music band, other open source projects, hobbies, etc.). Maybe we can come up with some principles to make it easy, not sure.

The weight of responsibilities didn't get easier with Russia's invasion of Ukraine. However, now I'm feeling more confident in the Leaflet community's capacity to help out with this. And, the need to urgently establish a sponsorship channel presented itself unexpectedly — [Microsoft FOSS fund just announced](https://twitter.com/sunnydeveloper/status/1521645029957070848) that Leaflet won its $10k grant for 2022, which would be silly to refuse. To move this forward, as a prerequisite, I had to set up the sponsorship systems — OpenCollective and GitHub Sponsors — anyway.

- [Leaflet on OpenCollective](https://opencollective.com/leafletjs) is up and running.
- [Leaflet on GitHub Sponsors](https://github.com/sponsors/Leaflet/) is in the process of approval from GitHub and should appear in a few days — Leaflet maintainers should have access to it now and make edits if necessary.

Once this is running, we'll have to establish some kind of governance system to distribute the funds to core contributors that actively work on Leaflet and its ecosystem. Let me know if you have any thoughts on how we should approach this. In addition, Emma from Microsoft Open Source Program indicated that they can help with this too — maybe I'll have some updates on this later.

Personally I won't accept any payouts from the Leaflet sponsorship fund as long as my employer (Mapbox) supports me spending time on Leaflet as part of my work responsibilities, but I'd love for people who work on Leaflet in their free time to be financially supported.

Remaining tasks (most can be discussed / worked on in parallel):

- [x] Get GitHub Sponsors approved
- [x] Receive the OSS grant.
- [ ] Settle on a governance model for distributing funds.
- [ ] Settle on the details of the public sponsorship pages — descriptions, funding tiers, potential rewards for higher tiers like putting a logo on the website / readme, etc.
- [x] Finally, add the "Sponsor" button to the repo, announce and promote the sponsorship opportunities.

cc @IvanSanchez @johnd0e @Falke-Design @Malvoz @jonkoops @perliedman @yohanboniface

**Comment by Falke-Design** [comment]
> [Microsoft FOSS fund just announced](https://twitter.com/sunnydeveloper/status/1521645029957070848) that Leaflet won its $10k grant for 2022

Wow that are greate news! 😄 If I can help you with something, just tell me 😉 

It would be a big benefit for the project if we can use parts of the sponsorship for browser simulations like https://www.browserstack.com/

**Comment by IvanSanchez** [comment]
I've glanced over #7704 and I feel like I should chip in.

So I'll make my position clear: I'm not a fan of Microsoft, at all. Besides considering myself a die-hard FLOSS activist (I could rant for hours about MS), there's the sin of creating MSIE. I think we all know that pain, from MSIE6's lack of `position:absolute` to MSIE11 `PointerEvent`s. This money is no redemption, and while I'll gladly take it, it cannot swerve my opinion.

The main reason I've kept away from github sponsorship is the fees - putting together credit card processing, plus stripe/paypal/opencollective fees, plus currency exchange fees, something like 10% can get lost in the way. Which to me feels... wrong, somehow. I hope in the future there'll be better mechanisms for this (I'd love EU funding, as https://joinup.ec.europa.eu/collection/fosseps/news/funding-sustainability hints).

---

> Settle on a governance model for distributing funds.

I'm not sure what the options for managing this will be, but since I've been freelancing and billing by the hour, my first naïve thought is that funds should be distributed by self-reported time spent of Leaflet stuff (as long as that time is not covered by an employer or a freelancer contract or the like). This implies a high level of trust in all other maintainers, which I think it'll be fine.

I don't think a "benevolent dictator" model is feasible at the moment - @mourner has a lot on his plate.

If anybody has a different idea on how to manage this... do speak out.

---

> It would be a big benefit for the project if we can use parts of the sponsorship for browser simulations like https://www.browserstack.com/

Apparently there should be no need - browser stack has its own sponsoring stuff going on at https://www.browserstack.com/open-source - but it needs applying and setting up.

**Reply by mourner** [reply]
> So I'll make my position clear: I'm not a fan of Microsoft, at all.

I'll respectfully disagree here — I think Microsoft, or at least their developer-related branches, has built a lot of good will over the last years that more than compensate its past mistakes, and it's worth supporting companies that got on the right track. Beyond their OSS support, I'm especially fond of how GitHub is moving after the acquisition, and also grateful to them for saving NPM which was near collapse at the time.

**Reply by Malvoz** [reply]
> > Settle on a governance model for distributing funds.
>
> I'm not sure what the options for managing this will be [...]

Something like [GitPay](https://github.com/worknenjoy/gitpay/blob/master/README.md) to automate the distribution would be cool (this is not an endorsement!). It'd be even cooler if this was something that Github supported, but that's [not yet the case](https://github.com/github/feedback/discussions/4517).

**Comment by jonkoops** [comment]
Just want to throw it out there that I am not looking for any financial compensation for my work on Leaflet. I do the limited amount of work I can out of my love for open-source software in my off time, and I will continue to do so.

**Comment by mourner** [comment]
Got the GitHub Sponsors approved: https://github.com/sponsors/Leaflet (also linked from the new "Sponsor" button on the repo)

**Reply by mdavids** [reply]
@SIDN Labs is happy to be sponsor number 4. Keep up the good work and peace for Ukraine!

**Reply by mourner** [reply]
@mdavids thank you so much ❤️
